src/testing/vortex/logged_process.zig

```diff
@@ -163,29 +103,35 @@ pub fn wait(
     self.expect_state_in(.{ .running, .terminated });
     defer self.expect_state_in(.{.terminated});
 
-    // Wait until the process runs to completion.
-    const term = self.child.wait();
+    const term = try self.child.wait();
+    self.state = .terminated;
+    return term;
+}
 
-    if (self.state() == .running) {
-        // Await thread in case this process is still running.
-        self.stdin_thread.join();
-    }
+/// If the process has exited, reap it and return the exit code.
+/// Otherwise, return null.
+pub fn wait_nonblocking(self: *LoggedProcess) ?u8 {
+    self.expect_state_in(.{ .running, .paused });
 
-    return term;
+    const result = std.posix.waitpid(self.child.id, std.posix.W.NOHANG);
```

Comment:
This is guaranteed to be race free, because `child.id` won't be re-used until we wait for is, right? 

I wonder why I didn't do the same for CFO? I guess that's because I wanted cfo to be cross platform, and didn't feel like writing the windows equivalent? Might be worth reconsidering!

I am also still divided in my mind on whether vortex should be Linux/POSIX only, or cross platform. One thing I realized is that now our upgrade test is cross-platform, and we had plans to move that to vortex... But maybe we shouldn't? 

Anyway, totally makes sense to continue vortex as Linux-first/only for know, we can always write more code later to make it cross platform!